### PR TITLE
[TF] Tensor.replacing: unit test and implicit broadcast

### PR DESCRIPTION
**Pull request by artste**
This pull request contains the following additions:
1) Unit test for tensor.replacing(with,where) fix (#24635)
1) Implicit broadcast for `with` parameter, with relative unit test.

Example:
```swift
// You can write this:
let modified = orig.replacing(with: Tensor<Float>(-1), where: orig.<3)

// Instead of this:
let modified_no_bst = orig.replacing(with: Tensor<Float>(-1).broadcast(like: orig), where: orig.<3)
```

### CODE REVIEW

File: test/TensorFlowRuntime/tensor.swift [hunk 1]
```diff
@@ -778,6 +778,25 @@ func testShapeGetter4() {
...
+@inline(never)
+func testReplacing1() {
+  let original = Tensor<Float>([[1.0, 0.0], [3.0, 0.0], [2.0, 3.0], [1.0, 0.0]])
+  let modified = original.replacing(with: Tensor<Float>(-1).broadcast(like: original), where: original.<3)
+  let expected = Tensor<Float>([[-1.0, -1.0], [3.0, -1.0], [-1.0, 3.0], [-1.0, -1.0]])
+  expectEqual(expected, modified)
+}
+TensorTests.testAllBackends("Replacing1", testReplacing1)
+
+@inline(never)
```
Comment (rxwei) [inline test/TensorFlowRuntime/tensor.swift]: No need to mark these as `@inline(never)` (some other tests were written this way for historical technical constraints). And could you use the trailing closure syntax instead? 

```swift
TensorTests.testAllBackends("replacing(with:where:) with broadcasting") {
   let original = Tensor<Float>([[1.0, 0.0], [3.0, 0.0], [2.0, 3.0], [1.0, 0.0]])
   let modified = original.replacing(with: Tensor<Float>(-1), where: original.<3)
   let expected = Tensor<Float>([[-1.0, -1.0], [3.0, -1.0], [-1.0, 3.0], [-1.0, -1.0]])
   expectEqual(expected, modified)
}
```

File: test/TensorFlowRuntime/tensor.swift [hunk 2]
```diff
@@ -778,6 +778,25 @@ func testShapeGetter4() {
...
+func testReplacing2() {
+  // Implicit broadcast
```
Comment (rxwei) [inline test/TensorFlowRuntime/tensor.swift]: This comment can be included in the test name, as demonstrated in the comment above.

File: test/TensorFlowRuntime/tensor.swift [hunk 3]
```diff
@@ -778,6 +778,25 @@ func testShapeGetter4() {
 }
 TensorTests.testAllBackends("ShapeGetter4", testShapeGetter4)
 
+@inline(never)
```
Comment (rxwei) [inline test/TensorFlowRuntime/tensor.swift]: Similarly, `@inline(never)` should be dropped and we should use the trailing closure syntax.

```swift
TensorTests.testAllBackends("replacing(with:where:)") {
   let original = Tensor<Float>([[1.0, 0.0], [3.0, 0.0], [2.0, 3.0], [1.0, 0.0]])
   let modified = original.replacing(with: Tensor<Float>(-1).broadcast(like: original), where: original.<3)
   let expected = Tensor<Float>([[-1.0, -1.0], [3.0, -1.0], [-1.0, 3.0], [-1.0, -1.0]])
   expectEqual(expected, modified)
}
```

Comment (rxwei) [reply test/TensorFlowRuntime/tensor.swift]: Also, could you please make sure these tests fit within 80 columns?

File: test/TensorFlowRuntime/tensor.swift (hunk 1 above)
Comment (artste) [reply test/TensorFlowRuntime/tensor.swift]: Great, this is way better.

**Comment by artste** [comment]
DONE:
+ Removed `@inline(never)`.
+ Refactor using trailing closure syntax.
+ All lines in 80 columns.

**Comment by rxwei** [comment]
@artste We've recently moved all TensorFlow APIs from apple/swift to [tensorflow/swift-apis](github.com/tensorflow/swift-apis). Would you mind moving this PR to that library instead? Sorry for the inconvenience!

**Comment by dan-zheng** [comment]
This PR seems inactive so I'll close it now. `TensorFlow` APIs have been moved to [https://github.com/tensorflow/swift-apis](https://github.com/tensorflow/swift-apis).

**Pull request closed without merging**